### DISCUSSION TITLE
CLI Feature: Recognize missing `mdx()` and prompt install

### DISCUSSION THREAD

**Discussion by tordans** [withastro/roadmap#420] [Proposal]
This is a request to make the error messages that the console provides more helpful / actionable.

---

I added mdx files to my astro repo recently and got this error message: 
> /some_path/some_file.mdx:2:5: **ERROR: Expected ";" but found ":"**

The [docs did not help](https://docs.astro.build/de/guides/integrations-guide/mdx/). Googling this is not easy.

Luckily from https://github.com/withastro/astro/issues/5658#issuecomment-1361372497 I learned, that I am likely missing the `mdx()` integration.

It would be great, if the CLI would analyze the error message, recognize that this is about an mdx file with this typical error message and that I don't have the mdx integration installed.

---

PS: Is cli the right term here? Or is this just console output? (Feel free to update the title…)

**Comment by bholmesdev** [comment]
This is a great suggestion @tordans. I've seen this issue enough times (and run into it _myself_ enough times) that a fix feels warranted! We can add a check for whether any Vite plugins have claimed `.mdx` files. If not, log an error recommending Astro's MDX integration.

**Comment by bholmesdev** [comment]
PR is up! Thanks for starting this convo @tordans 🙌 
https://github.com/withastro/rfcs/discussions/420
